tests/delete/tests.py

```diff
@@ -90,6 +90,30 @@ def test_protect_multiple(self):
         with self.assertRaisesMessage(ProtectedError, msg):
             a.protect.delete()
 
+    def test_protect_chained(self):
+        p = P.objects.create()
+        r = R.objects.create(p=p)
+        B.objects.create(protect=r)
```

Comment:
We can use `create_a()` hook, e.g.
```python
    def test_protect_path(self):
        a = create_a('protect')
        a.protect.p = P.objects.create()
        a.protect.save()
        msg = (
            "Cannot delete some instances of model 'P' because they are "
            "referenced through protected foreign keys: 'R.p'.",
        )   
        with self.assertRaisesMessage(ProtectedError, msg):
            a.protect.p.delete()
```